smurfs/src/reducers/reducers.js

```diff
@@ -0,0 +1,51 @@
+import { FETCH_SMURFS_START, FETCH_SMURFS_SUCCESS, FETCH_SMURFS_FAILURE, ADD_SMURFS, REMOVE_SMURFS } from "../actions/actions";
+export const initialState = {
+    smurfs: [{
+        name: "",
+        age: "",
+        height: "",
+        id: ""
+    }],
+    isFetching: false,
+    error: ""
+};
+
+const reducer = (state = initialState, action) => {
+    switch (action.type) {
+        case FETCH_SMURFS_START:
+            return {
+                ...state,
+                isFetching: true
+            };
+        case FETCH_SMURFS_SUCCESS:
+            return {
+                ...state,
+                smurfs: action.payload,
+                isFetching: false,
+                error: ''
+            };
+        case FETCH_SMURFS_FAILURE:
+            return {
+                ...state,
+                isFetching: false,
+                error: action.payload
+            };
+        case ADD_SMURFS:
+            return {
+                ...state,
+                smurfs: [...state.smurfs, action.payload]
+            };
+        case REMOVE_SMURFS:
+            return {
+                ...state,
+                smurfs: state.smurfs.filter((smurf) => !smurf.id)
+
```

Comment:
So so so close with this REMOVE_SMURFS! Just remember that when doing the filter you have to compare something. Right now you are just returning any smurf who doesnt have an id(all smurfs have an id so this will return an empty array)